interfaces/zookeeper_client/v0/schema.py

```diff
@@ -0,0 +1,92 @@
+"""This file defines the schemas for the provider and requirer sides of the zookeeper_client interface.
+
+It must expose two interfaces.schema_base.DataBagSchema subclasses called:
+- ProviderSchema
+- RequirerSchema
+"""
+
+from typing import Annotated
+
+from interface_tester.schema_base import DataBagSchema
+from pydantic import (
+    AfterValidator,
+    BaseModel,
+    Field,
+    IPvAnyAddress,
+    conint,
+)
+
+
+class URI(BaseModel):
+    ip: IPvAnyAddress
+    port: conint(ge=0, le=65535) | None
+
+    def __init__(self, value: str) -> None:
+        ip, _, port = value.partition(":")
+        if not port:
+            port = None
+        super().__init__(ip=ip, port=port)
+
+
+def parse_endpoints(value: str):
+    uris, _, znode = value.partition("/")
+    [URI(val) for val in uris.split(",")]
+
+
+Endpoints = Annotated[str, AfterValidator(parse_endpoints)]
+
+
+class ZookeeperProviderAppData(BaseModel):
+    database: str = Field(
+        description="The parent chroot zNode granted to the requirer",
+        examples=["/myAppb"],
+        title="zNode",
+    )
+
+    endpoints: Endpoints = Field(
+        description="A comma-seperated list of ZooKeeper server uris, and parent chroot zNode",
+        examples=["10.141.78.133:2181,10.141.78.50:2181,10.141.78.45:2181/myappB"],
```

Comment:
Is there any reason for the `myAppb` capitalization to be different from the database value (`myAppb`)?

Reply:
Nope, fixed the discrepancy in ba07bb21c16ff0732c456d43dfd1a8bb1bd17576.
Nice job spotting that :eagle: